nixos/modules/services/networking/adguardhome.nix

```diff
@@ -12,24 +13,16 @@ let
     "--config /var/lib/AdGuardHome/AdGuardHome.yaml"
   ] ++ cfg.extraArgs);
 
-  configFile = pkgs.writeTextFile {
-    name = "AdGuardHome.yaml";
-    text = builtins.toJSON cfg.settings;
-    checkPhase = "${pkgs.adguardhome}/bin/adguardhome -c $out --check-config";
-  };
-  defaultBindPort = 3000;
+  settings = if (cfg.settings != null)
+    then cfg.settings // {
+      http.address = "${cfg.host}:${toString cfg.port}";
```

Comment:
I just noticed that this line might break configs where an explicit older `schema_version` is in place, which did not use this config style.

Reply:
My suggestion would be something like:
```nix
  settings = if (cfg.settings != null)
             then cfg.settings //
                  (if cfg.settings.schema_version < 23 then
                    { bind_host = cfg.host; bind_port = toString cfg.port;}
                   else { http.address = "${cfg.host}:${toString cfg.port}"; })
             else null;
```

Maybe it makes sense to add back a single test case with predetermined `schema_version`.

Reply:
The problem is the test would still fail, the only way to get this to work is if I backport to an older version for this test. I'll try to do this.

Reply:
My idea for the test case was a bit different, I've added a comment below. Sorry for the miscommunication.